EIPS/eip-2930.md

```diff
@@ -91,6 +91,8 @@ def process_access_list(access_list) -> Tuple[List[Set[Address], Set[Pair[Addres
 
 The access list is NOT charged per-byte fees like tx data is; the per-item costs described above are meant to cover the bandwidth costs of the access list data in addition to the costs of accessing those accounts and storage keys when evaluating the transaction.
 
+Additionally, the `chainId` field must be equal to either the chain id of the current network or `0` if it is unprotected.
```

Comment:
Is the `chainId` rlp format defined somewhere? It can either be an arbitrarily large integer (in which case the empty value is one byte only), or e.g. a `uint64`, in which case it's fixed size (and quite largre). So I suppose an arbitrary-size integer would make the most sense. 
Also, having `0` be unprotected is a pretty important piece of consensus-vital information, so it should not be mentioned as a mere afterthought. Maybe make a bold "Note" or move it up or something, to make it obvious. A consequence would also be (I guess) that `V` is `27` or `28` (or `0` and `1`?)  

Also, under "testcases", we could add some RLP-encoded blobs of txs

- With `unprotected` transaction, rlp + signing hash 
- With goerli-chainid transaction, rlp + signing hash

That would also clarify the `V` domain. 

@vbuterin thoughts?

Reply:
I'm not 100% certain we should use `0` to denote unprotected txs. Do we really want unprotected txs to be "a thing"? It existed earlier for historic reasons, but since only new signers will be able to sign these, we have the option now to get rid of it.

Reply:
I think @MicahZoltu had some reasoning behind why we'd still want to have unprotected transactions. However, I lean on the side of removing the functionality completely.

Reply:
I don't believe `chainId` is defined anywhere, but I agree that an arbitrarily large integer makes the most sense.

Reply:
Unprotected transactions allow for trustless contract deployments across multiple chains which I believe to be of significant value as it makes testing and integration *way* easier when a particular contract (e.g., Uniswap) lives at the same address on every testnet and mainnet.

Reply:
> Unprotected transactions allow for trustless contract deployments across multiple chains

But isn't the "correct" way to do this to just use CREATE2? Just use a single old-style transaction to make a standardized contract factory and then use that to make any desired fixed-address contracts.

Reply:
I'm happy with separating out chainid from the v value; it definitely is cleaner to have each element represent a separate thing than to have an awkward de-facto mash of integers and bitfields.

Reply:
If we're in agreement, I'll remove the caveat that 2930 txs are unprotected when `chainId == 0`.

Reply:
As long as there is a way to deploy a factory without a chain ID then create 2 will work. This does mean that we won't be able to use access lists for everything though, since at least that factory deployment transaction won't be able to use access lists. In that case though, the access list would only have contained the account info for the deployer.